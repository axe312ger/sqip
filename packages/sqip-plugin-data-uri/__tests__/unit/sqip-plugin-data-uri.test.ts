import sqipPluginDataUri from '../../src/sqip-plugin-data-uri'

import { Swatch } from '@vibrant/color'

const EXAMPLE_SVG = Buffer.from(
  '<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 100 100"><circle cx="50" cy="50" r="40" fill="red" stroke="#000" stroke-width="3"/></svg>'
)

const mockedMetadata = {
  height: 0,
  width: 0,
  originalHeight: 0,
  originalWidth: 0,
  palette: {
    DarkMuted: new Swatch([4, 2, 0], 420),
    DarkVibrant: new Swatch([4, 2, 1], 421),
    LightMuted: new Swatch([4, 2, 2], 422),
    LightVibrant: new Swatch([4, 2, 3], 423),
    Muted: new Swatch([4, 2, 4], 424),
    Vibrant: new Swatch([4, 2, 5], 425)
  },
  type: 'mocked'
}
const mockedConfig = {
  input: 'mocked',
  output: 'mocked',
  plugins: ['pixels']
}

describe('sqip-plugin-data-uri', () => {
  it('does not touch svg', () => {
    const plugin = new sqipPluginDataUri({
      pluginOptions: {},
      options: {},
      metadata: mockedMetadata,
      sqipConfig: mockedConfig
    })
    const result = plugin.apply(EXAMPLE_SVG)
    expect(result).toBe(EXAMPLE_SVG)
  })
  it('encodes in mini-svg-data-uri and base64', () => {
    const metadata = { ...mockedMetadata }
    const plugin = new sqipPluginDataUri({
      pluginOptions: {},
      options: {},
      metadata,
      sqipConfig: mockedConfig
    })
    plugin.apply(EXAMPLE_SVG)
    expect(metadata).toMatchSnapshot()
  })
})
