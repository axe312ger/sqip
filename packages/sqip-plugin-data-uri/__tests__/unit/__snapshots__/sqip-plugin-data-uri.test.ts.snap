// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sqip-plugin-data-uri encodes in mini-svg-data-uri and base64 1`] = `
Object {
  "dataURI": "data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 100 100'%3e%3ccircle cx='50' cy='50' r='40' fill='red' stroke='black' stroke-width='3'/%3e%3c/svg%3e",
  "dataURIBase64": "data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZpZXdCb3g9IjAgMCAxMDAgMTAwIj48Y2lyY2xlIGN4PSI1MCIgY3k9IjUwIiByPSI0MCIgZmlsbD0icmVkIiBzdHJva2U9IiMwMDAiIHN0cm9rZS13aWR0aD0iMyIvPjwvc3ZnPg==",
  "height": 0,
  "originalHeight": 0,
  "originalWidth": 0,
  "palette": Object {
    "DarkMuted": Object {
      "population": 420,
      "rgb": Array [
        4,
        2,
        0,
      ],
    },
    "DarkVibrant": Object {
      "population": 421,
      "rgb": Array [
        4,
        2,
        1,
      ],
    },
    "LightMuted": Object {
      "population": 422,
      "rgb": Array [
        4,
        2,
        2,
      ],
    },
    "LightVibrant": Object {
      "population": 423,
      "rgb": Array [
        4,
        2,
        3,
      ],
    },
    "Muted": Object {
      "population": 424,
      "rgb": Array [
        4,
        2,
        4,
      ],
    },
    "Vibrant": Object {
      "population": 425,
      "rgb": Array [
        4,
        2,
        5,
      ],
    },
  },
  "type": "mocked",
  "width": 0,
}
`;
